### PR TITLE
Unified fees tx

### PR DESCRIPTION
**Pull request by lucaslopezf**
- Unified fees tx
- Refector createExternal decoding
- Consolidate all addresses to robust format

:link: [zboto Link](https://app.clickup.com/t/8693r1ued)

### CODE REVIEW

File: factory.go [hunk 1]
```diff
@@ -173,7 +173,7 @@ func (p *FilecoinParser) ParseGenesis(genesis *types.GenesisBalances, genesisTip
 			Level:       0,
 			TxTimestamp: genesisTimestamp,
 			TxTo:        balance.Key,
-			Amount:      amount.Int,
+			Amount:      amount.Int.String(),
```
Comment (emmanuelm41) [inline factory.go:176]: This was not really the idea. It is better to keep this as a BigInt. We want to change it on the Beryx API (convert from BigInt to String before sending the response back as JSON)

File: parser/v1/parser.go [hunk 2]
```diff
@@ -107,7 +107,7 @@ func (p *Parser) ParseTransactions(traces []byte, tipset *types.ExtendedTipSet,
 				TxFrom:      trace.Msg.From.String(),
 				TxTo:        trace.Msg.To.String(),
 				TxType:      txType,
-				Amount:      trace.Msg.Value.Int,
+				Amount:      trace.Msg.Value.Int.String(),
```
Comment (emmanuelm41) [inline parser/v1/parser.go:110]: Same here!

File: parser/types.go [hunk 3]
```diff
@@ -78,8 +78,13 @@ type BurnFee struct {
...
 }
 
+type FeeData struct {
+	FeesMetadata
+	Amount string `json:"amount"`
+}
+
 type FeesMetadata struct {
-	TxType                string
+	TxType                string `json:"tx_type"`
```
Comment (emmanuelm41) [inline parser/types.go:87]: TxType is here, but is actually never filled up when instantiating this struct

File: parser/v2/parser.go [hunk 4]
```diff
@@ -93,17 +96,11 @@ func (p *Parser) ParseTransactions(traces []byte, tipset *types.ExtendedTipSet,
...
 			if len(subTxs) > 0 {
 				transactions = append(transactions, subTxs...)
 			}
 		}
-
-		// Fees
-		if trace.GasCost.TotalCost.Uint64() > 0 {
-			feeTx := p.feesTransactions(trace, tipset, transaction.TxType, transaction.Id)
-			transactions = append(transactions, feeTx)
-		}
```
Comment (emmanuelm41) [inline parser/v2/parser.go:106]: Fees were created only at level 0 tx. I think new changes attempt to add those values on subcalls too. I guess we should not. Only level 0 tx should have fees data field with some real data

File: parser/v2/parser.go [hunk 5]
```diff
@@ -231,65 +233,43 @@ func (p *Parser) parseTrace(trace typesV2.ExecutionTraceV2, mainMsgCid cid.Cid,
...
+				Amount:       gasCost.MinerTip.String(),
+			},
+			OverEstimationBurnFee: parser.OverEstimationBurnFee{
+				BurnAddress: parser.BurnAddress,
+				Amount:      gasCost.OverEstimationBurn.String(),
+			},
+			BurnFee: parser.BurnFee{
+				BurnAddress: parser.BurnAddress,
+				Amount:      gasCost.BaseFeeBurn.String(),
+			},
```
Comment (emmanuelm41) [inline parser/v2/parser.go:265]: Is there any other info we can add here?

**Review by emmanuelm41 — CHANGES_REQUESTED** [review]
Left some important comments

File: parser/v2/parser.go (hunk 5 above)
Comment (lucaslopezf) [reply parser/v2/parser.go:265]: No, just the amount, because the other information such as ID and parent_id is irrelevant for us I think

**Pull request closed without merging**